h2o-core/src/main/java/water/util/ComparisonUtils.java

```diff
@@ -0,0 +1,192 @@
+                Difference difference = new Difference(description, first, second);
+                addDifference(difference);
+            }
+        }
+
+        private static boolean compareValuesUpToTolerance(float[] first, float[] second, double relativeTolerance) {
+            if (first == null) {
+                return second == null;
+            }
+            if (first.length != second.length) {
+                return false;
+            }
+            for (int i = 0; i < first.length; i++) {
+                if(!compareValuesUpToTolerance(first[i], second[i], relativeTolerance)) return false;
+            }
+            return true;
+        }
+
+        public void compareValuesUpToTolerance(String description, double[] first, double[] second) {
+            if (_ignoredFields.contains(description)) return;
+            if (!compareValuesUpToTolerance(first, second, _relativeTolerance)) {
+                String firstString = Arrays.toString(first);
+                String secondString = Arrays.toString(second);
+                Difference difference = new Difference(description, firstString, secondString);
+                addDifference(difference);
+            }
+        }
+
+        private static boolean compareValuesUpToTolerance(double[] first, double[] second, double relativeTolerance) {
+            if (first == null) {
+                return second == null;
+            }
+            if (first.length != second.length) {
+                return false;
+            }
+            for (int i = 0; i < first.length; i++) {
+                if(!compareValuesUpToTolerance(first[i], second[i], relativeTolerance)) return false;
+            }
+            return true;
+        }
+
+        public void compareValuesUpToTolerance(String description, float[] first, float[] second) {
+            if (_ignoredFields.contains(description)) return;
+            if (!compareValuesUpToTolerance(first, second, _relativeTolerance)) {
+                String firstString = Arrays.toString(first);
+                String secondString = Arrays.toString(second);
+                Difference difference = new Difference(description, firstString, secondString);
+                addDifference(difference);
+            }
+        }
+
+        private static boolean compareValuesUpToTolerance(double first, double second, double relativeTolerance) {
+            if (Double.isNaN(first)) {
+                return Double.isNaN(second);
+            }
+
+            if (first == Double.POSITIVE_INFINITY) {
+                return second == Double.POSITIVE_INFINITY;
+            }
+
+            if (first == Double.NEGATIVE_INFINITY) {
+                return second == Double.NEGATIVE_INFINITY;
+            }
+
+            // covers scenario when both are 0.0.
+            if (first == second) {
+                return true;
+            }
+
+            double ratio = first / second;
```

Comment:
this makes the comparison asymmetric and makes "second" more important 

your difference formula essentially looks like 

abs((first - second) / second)

for this purpose, I would use 

abs(first - second) / max(abs(first), abs(second))

there are also other ways - https://en.wikipedia.org/wiki/Relative_change_and_difference

Reply:
Good point, thanks!